ngraph/core/reference/include/ngraph/coordinate_transform.hpp

```diff
@@ -31,11 +31,18 @@ namespace ngraph
     ///             {1,0}, {1,1}, {2,2}
     class CoordinateIterator
     {
-    public:
         /// \brief Coordinates iterator constructor
         /// \param target_shape The target shape for coordinates iteration
         /// \param is_end The flag indicates that the coordinate iterator is the last.
-        CoordinateIterator(const Shape& target_shape, bool is_end = false);
+        CoordinateIterator(const Shape& target_shape, bool is_end);
```

Comment:
You moved this constructor to private. Do we still have cases where `is_end` is true?

Reply:
Yes we have: https://github.com/openvinotoolkit/openvino/blob/2b70fa14732d3d5740b4525ad38a6877f2f86959/ngraph/core/reference/src/coordinate_transform.cpp#L537